src/peft/peft_model.py

```diff
@@ -177,7 +177,12 @@ def from_pretrained(cls, model, model_id, adapter_name="default", is_trainable=F
 
         # load the config
         config = PEFT_TYPE_TO_CONFIG_MAPPING[
-            PeftConfig.from_pretrained(model_id, subfolder=kwargs.get("subfolder", None), **kwargs).peft_type
+            PeftConfig._get_peft_type(
+                model_id,
+                subfolder=kwargs.get("subfolder", None),
+                revision=kwargs.get("revision", None),
+                cache_dir=kwargs.get("cache_dir", None),
+            )
         ].from_pretrained(model_id, subfolder=kwargs.get("subfolder", None), **kwargs)
```

Comment:
This line too needs update as it is currently taking all kwargs?

Reply:
This shouldn't be needed as the kwargs will be dealt properly inside the config class that will separate the kwargs into 3 sub-kwargs `hf_hub_download_kwargs`, `class_kwargs` (for overriding class attributes) and `other_kwargs`

Reply:
I meant `].from_pretrained(model_id, subfolder=kwargs.get("subfolder", None), **kwargs)` this line is still using all kwargs